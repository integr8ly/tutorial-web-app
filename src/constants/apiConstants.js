const API_RESPONSE_AUTH_USERNAME = 'username';
const API_RESPONSE_AUTH_EMAIL = 'email';

const apiTypes = {
  API_RESPONSE_AUTH_USERNAME,
  API_RESPONSE_AUTH_EMAIL
};

export default apiTypes;
