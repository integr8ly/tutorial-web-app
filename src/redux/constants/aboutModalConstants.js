const ABOUT_MODAL_SHOW = 'ABOUT_MODAL_SHOW';
const ABOUT_MODAL_HIDE = 'ABOUT_MODAL_HIDE';

export { ABOUT_MODAL_SHOW, ABOUT_MODAL_HIDE };
