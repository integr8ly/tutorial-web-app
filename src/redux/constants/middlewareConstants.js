const LIST_WALKTHROUGH = 'LIST_WALKTHROUGH';
const CREATE_WALKTHROUGH = 'CREATE_WALKTHROUGH';
const REMOVE_WALKTHROUGH = 'REMOVE_WALKTHROUGH';
const GET_AMQ_CREDENTIALS = 'GET_AMQ_CREDENTIALS';
const GET_ENMASSE_CREDENTIALS = 'GET_ENMASSE_CREDENTIALS';
const GET_PROVISIONING_USER = 'GET_PROVISIONING_USER';
const GET_CUSTOM_CONFIG = 'GET_CUSTOM_CONFIG';
const PROVISION_SERVICE = 'PROVISION_SERVICE';
const GET_RHMICONFIG_CR = 'GET_RHMICONFIG_CR';

const SERVICE_TYPES = {
  PROVISIONED_SERVICE: 'PROVISIONED_SERVICE'
};

const SERVICE_STATUSES = {
  PROVISIONING: 'PROVISIONING',
  PROVISIONED: 'PROVISIONED',
  DELETING: 'DELETING',
  UNAVAILABLE: 'UNAVAILABLE'
};

export {
  CREATE_WALKTHROUGH,
  REMOVE_WALKTHROUGH,
  LIST_WALKTHROUGH,
  GET_AMQ_CREDENTIALS,
  GET_ENMASSE_CREDENTIALS,
  GET_PROVISIONING_USER,
  GET_CUSTOM_CONFIG,
  PROVISION_SERVICE,
  SERVICE_STATUSES,
  SERVICE_TYPES,
  GET_RHMICONFIG_CR
};
