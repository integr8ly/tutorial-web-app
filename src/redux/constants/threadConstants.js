const GET_THREAD = 'GET_THREAD';

export { GET_THREAD };
