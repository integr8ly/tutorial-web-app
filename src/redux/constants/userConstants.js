const USER_CREATE = 'USER_CREATE';
const USER_DELETE = 'USER_DELETE';
const USER_INFO = 'USER_INFO';
const USER_LOGIN = 'USER_LOGIN';
const USER_LOGOUT = 'USER_LOGOUT';
const USER_STORED_DATA = 'USER_STORED_DATA';
const USER_STORED_DATA_REMOVE = 'USER_STORED_DATA_REMOVE';

export { USER_CREATE, USER_DELETE, USER_INFO, USER_LOGIN, USER_LOGOUT, USER_STORED_DATA, USER_STORED_DATA_REMOVE };
