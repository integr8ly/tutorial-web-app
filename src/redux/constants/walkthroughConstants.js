const LIST_WALKTHROUGH = 'LIST_WALKTHROUGH';
const CREATE_WALKTHROUGH = 'CREATE_WALKTHROUGH';
const REMOVE_WALKTHROUGH = 'REMOVE_WALKTHROUGH';

export { CREATE_WALKTHROUGH, REMOVE_WALKTHROUGH, LIST_WALKTHROUGH };
