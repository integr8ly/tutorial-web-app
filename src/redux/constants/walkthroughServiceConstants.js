const ADD_RESOURCE = 'ADD_RESOURCE';
const REMOVE_RESOURCE = 'REMOVE_RESOURCE';

export { ADD_RESOURCE, REMOVE_RESOURCE };
