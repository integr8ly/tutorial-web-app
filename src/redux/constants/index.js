import * as aboutModalTypes from './aboutModalConstants';
import * as userTypes from './userConstants';
import * as threadTypes from './threadConstants';
import * as walkthroughTypes from './walkthroughConstants';

const reduxTypes = {
  aboutModalTypes,
  userTypes,
  threadTypes
};

export { reduxTypes as default, reduxTypes, aboutModalTypes, userTypes, threadTypes, walkthroughTypes };
