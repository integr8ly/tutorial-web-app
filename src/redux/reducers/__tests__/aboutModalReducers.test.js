import { aboutModalReducers } from '..';
import { aboutModalTypes as types } from '../../constants';

describe('ConfirmationModalReducers', () => {
  it('should return the initial state', () => {
    expect(aboutModalReducers.initialState).toBeDefined();
  });

  it('should handle all defined types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = aboutModalReducers(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot('defined types');
    });
  });
});
