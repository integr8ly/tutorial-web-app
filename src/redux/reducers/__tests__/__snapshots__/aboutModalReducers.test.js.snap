// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModalReducers should handle all defined types: defined types 1`] = `
Object {
  "result": Object {
    "show": true,
    "type": "ABOUT_MODAL_SHOW",
  },
  "type": "ABOUT_MODAL_SHOW",
}
`;

exports[`ConfirmationModalReducers should handle all defined types: defined types 2`] = `
Object {
  "result": Object {
    "show": false,
  },
  "type": "ABOUT_MODAL_HIDE",
}
`;
